### PR TITLE
test(es2016): dynamic import()

### PR DESCRIPTION
**Pull request by ghost**
> `import()` is proposed to work in both scripts and modules. This gives script code an easy asynchronous entry point into the module world, allowing it to start running module code.

> This proposal would be a small expansion of the existing JavaScript and HTML capabilities, using the same framework of specifying syntactic forms in the JavaScript specification, which delegate to the host environment for their heavy lifting. HTML's infrastructure for fetching and resolving modules would be leveraged to define its side of the story. Similarly, Node.js would supply its own definitions for HostPrepareImportedModule and HostResolveImportedModule to make this proposal work there.

**Citing:** https://github.com/tc39/proposal-dynamic-import
**Spec:** https://tc39.github.io/proposal-dynamic-import/
**See also:** https://github.com/kangax/compat-table/issues/1034, https://github.com/kangax/compat-table/issues/316, https://github.com/nodejs/node-eps/pull/39

### CODE REVIEW

File: data-es2016plus.js [hunk 1]
```diff
@@ -368,6 +368,101 @@ exports.tests = [
...
+    name: 'import()',
+    categpry: '2017 features',
+    significance: 'small',
+    spec: 'https://tc39.github.io/proposal-dynamic-import/',
+    subtests: [
+      {
+        name: 'import',
+        exec: function () {/*
+          try {
+            return typeof import === "function";
```
Comment (ljharb) [inline data-es2016plus.js:380]: this is false; `import()` is strictly **not** a function, it's a keyword followed by parens.

File: data-es2016plus.js [hunk 2]
```diff
@@ -368,6 +368,101 @@ exports.tests = [
...
+          return import("./foo.js") instanceof Promise;
+        */},
+        res: {
+        }
+      },
+      {
+        name: 'throw',
+        exec: function () {/*
+          try {
+            import("")
```
Comment (ljharb) [inline data-es2016plus.js:401]: wouldn't `import('')` potentially reject anyways, since that's not a valid path?

**Review by ljharb — CHANGES_REQUESTED** [review]

**Comment by ljharb** [comment]
@domenic i'd love your input here on what things this should be testing

File: data-es2016plus.js (hunk 2 above)
Comment (domenic) [reply data-es2016plus.js:401]: That's up to the host environment. Testing import() is equivalent to testing import declarations...

File: data-es2016plus.js [hunk 3]
```diff
@@ -368,6 +368,101 @@ exports.tests = [
...
+            });
+        */},
+        res: {
+        }
+      },
+      {
+        name: 'throws without params',
+        exec: function () {/*
+          try {
+            import();
```
Comment (domenic) [inline data-es2016plus.js:443]: This is grammatically disallowed, not runtime disallowed, so it needs an eval wrapper.

File: data-es2016plus.js [hunk 4]
```diff
@@ -368,6 +368,101 @@ exports.tests = [
...
+            return true;
+          }
+        */},
+        res: {
+        }
+      },
+      {
+        name: 'rejects without params',
+        exec: function () {/*
+          var p = import();
```
Comment (domenic) [inline data-es2016plus.js:455]: This test is invalid

**Comment by ljharb** [comment]
I'm pretty sure this feature should not be tested at all except in a Modules tab (that does not yet exist), along with import/export itself.

**Comment by ljharb** [comment]
Closing per https://github.com/kangax/compat-table/issues/316#issuecomment-252405375

**Comment by jhabdas** [comment]
Lame

**Comment by ljharb** [comment]
I'm going to delete your inappropriate, ableist comment. Please be respectful.

**Comment by yukulele** [comment]
@ljharb is modules tab planed?

**Comment by ljharb** [comment]
Loosely yes.

**Comment by ghost** [comment]
@yukulele it's an unnecessary complexity barrier throwing off the validity of the Node charts—very few people are aware of this

**EDIT:** Opinions my own.™

**Pull request closed without merging**